website/versioned_docs/version-5.x/ecosystem-alpinejs.md

```diff
@@ -0,0 +1,236 @@
+export const unmount = alpinejslifecycles.unmount;
+```
+
+#### Via cdn
+
+Example usage when installed via CDN:
+
+```js
+const alpinejsApp = window.singleSpaAlpinejs.default({
+  template: `
+    <div class="rounded overflow-hidden shadow-lg font-sans p-1 m-1" 
+         x-data="{ open: false }">
+      <div class="font-bold p-1">Example for x-show attribute</div>
+      <button class="bg-transparent hover:bg-blue-500 text-blue-700 font-semibold 
+      hover:text-white py-2 px-4 border border-blue-500 
+      hover:border-transparent rounded" @click="open = !open">Open/Close</button>
+      <div x-show="open" class="text-4xl">
+          Hey, I'm open
+      </div>
+    </div>`,
+});
+
+singleSpa.registerApplication({
+  name: 'name',
+  app: alpinejsApp,
+  activeWhen: () => true,
+});
+```
+
+### _2 - Template with externally defined `x-data`_
+
+- You could also provide `x-data` externally and the helper will add it to the component.
+  - The `x-data` can be provided in the following forms (via the options attribute `xData`)
+    - an object
+    - a function
+    - a function returning a promise.
+
+#### Via npm
+
+```js
+import singleSpaAlpinejs from 'single-spa-alpinejs';
+const alpinejslifecycles = singleSpaAlpinejs({
+  template: `
+    <div class="rounded overflow-hidden shadow-lg font-sans p-1 m-1">
+      <div class="font-bold p-1">Example for x-show attribute</div>
+      <button class="bg-transparent hover:bg-blue-500 text-blue-700 font-semibold 
+      hover:text-white py-2 px-4 border border-blue-500 
+      hover:border-transparent rounded" @click="open = !open">Open/Close</button>
+      <div x-show="open" class="text-4xl">
+          Hey, I'm open
+      </div>
+    </div>`,
+  xData: { open: false },
+});
+
+export const bootstrap = alpinejslifecycles.bootstrap;
+export const mount = alpinejslifecycles.mount;
+export const unmount = alpinejslifecycles.unmount;
+```
+
+#### Via cdn
+
+Example usage when installed via CDN:
+
+```js
+const alpinejsApp = window.singleSpaAlpinejs.default({
+  template: `
+    <div class="rounded overflow-hidden shadow-lg font-sans p-1 m-1">
+      <div class="font-bold p-1">Example for x-show attribute</div>
+      <button class="bg-transparent hover:bg-blue-500 text-blue-700 font-semibold 
+      hover:text-white py-2 px-4 border border-blue-500 
+      hover:border-transparent rounded" @click="open = !open">Open/Close</button>
+      <div x-show="open" class="text-4xl">
+          Hey, I'm open
+      </div>
+    </div>`,
+  xData: { open: false },
+});
+
+singleSpa.registerApplication({
+  name: 'name',
+  app: alpinejsApp,
+  activeWhen: () => true,
+});
+```
+
+### _3 - Template with externally defined `x-data` with `x-init`_
+
+- You can also provide `x-init` externally along with the `x-data` and the helper will add it to the component.
+
+- The `x-init` can be provided in the following forms (via the options attribute `xInit`)
+  - a function
```

Comment:
Does the return value of the function matter?

Reply:
agreed .. will change it to just say it needs to be a function